ruddock/auth_utils.py

```diff
@@ -187,12 +187,18 @@ def hash_password(password, salt, rounds, algorithm):
   Algorithms using the passlib library are returned in base64 format.
   Algorithms using the hashlib library are returned in hex format.
   """
+  # Dirty hack to get Py3 to work properly. The hashlib functions we use here
+  # require its inputs to be 'bytes', not 'str'.
+  if type(password) is str:
+    password = str.encode(password)
+  if type(salt) is str:
+    salt = str.encode(salt)
   if algorithm == 'pbkdf2_sha256':
     # Rounds must be set.
     if rounds is None:
       return None
     result = hashlib.pbkdf2_hmac('sha256', password, salt, rounds)
```

Comment:
instead of making password and salt be bytestrings, why not just do this:

```
result = hashlib.pbkdf2_hmac('sha256', password.encode(), salt.encode(), rounds)
```